### ISSUE TITLE
Add SiLU activation / swish function

### ISSUE THREAD

**Issue by antimora** [burn-rs/burn#462]
### Feature description

The feature being requested is the implementation of the Sigmoid Linear Unit (SiLU) function, also known as the swish function. It should be applied element-wise. The mathematical expression for SiLU is given by silu(x) = x*σ(x), where σ(x) is the logistic sigmoid function.

### Feature motivation

The SiLU function is used in the Conformer model, which is a model that has gained popularity due to its effectiveness.

### Suggest a Solution

The solution should resemble PyTorch's implementation of the SiLU function. More details can be found in the [PyTorch documentation](https://pytorch.org/docs/stable/generated/torch.nn.SiLU.html).
